static/app/components/feedback/widget/useFeedbackForm.tsx

```diff
@@ -0,0 +1,22 @@
+import {useCallback} from 'react';
+
+import {
+  useFeedback,
+  type UseFeedbackOptions,
+} from 'sentry/components/feedback/widget/useFeedback';
+
+export function useFeedbackForm(props: UseFeedbackOptions) {
+  const {feedback, options} = useFeedback(props);
+
+  const openForm = useCallback(async () => {
+    if (!feedback) {
+      return;
+    }
+
+    const form = await feedback.createForm(options);
+    form.appendToDom();
+    form.open();
+  }, [feedback, options]);
+
+  return {feedback, openForm};
```

Comment:
probably also want to cleanup the form sometime too right?

Reply:
Does the form need to be cleaned up? It should be removed from the dom when the user closes it, right?

Reply:
it'll render with `open={false}` so there'll be an empty div hanging around, but the object pointing to it with all the methods should be eligible for GC

Reply:
Ah yes I do see it leaving behind a div. I agree, added a cleanup on unmount. Will make using it in my use case a bit more difficult, but makes sense.

 I could actually see this hook living at the top level of our app and all consumers would use the same form instance, actually. It would work similarly to our global modal component then.

Reply:
> actually see this hook living at the top level of our app and all consumers would use the same form instance, actually. It would work similarly to our global modal component then.

We could probably set that up now: do like a Provider that is like `<context.Provider value={{openForm, removeForm}}>{children}</context.Provider>` and then this `useFeedbackForm()` is the Consumer for that. Then any callsites in the app would share the same `formRef`